app/assets/javascripts/discourse/app/components/d-navigation.hbs

```diff
@@ -1,19 +1,36 @@
-<BreadCrumbs
-  @categories={{this.categories}}
-  @category={{this.category}}
-  @noSubcategories={{this.noSubcategories}}
-  @tag={{this.tag}}
-  @additionalTags={{this.additionalTags}}
-/>
+{{#if this.isQueryFilterMode}}
+  <div class="topic-query-filter">
+    <Input
+      class="topic-query-filter__input"
+      @value={{this.queryString}}
+      @enter={{route-action "changeQueryString" this.queryString}}
```

Comment:
Maybe we don't need changeQueryString action and can we use `mut` https://api.emberjs.com/ember/4.10/classes/Ember.Templates.helpers/methods/mut?anchor=mut

Reply:
We need this route-action here because we're changing the `q` property of the `discovery.filter` controller which is not available from this component.